### PR TITLE
Add a command to switch output format in cli

### LINKED ISSUE #1267: allow datafusion-cli to switch different output format within CLI
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

allow datafusion-cli to switch different output format within CLI

**Describe the solution you'd like**

using command that starts with `\`

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by capkurmagati**
fixes #1267

# Which issue does this PR close?

Closes #.

 # Rationale for this change

# What changes are included in this PR?

```
DataFusion CLI v5.1.0-SNAPSHOT

❯ \?
+------------------+-----------------------------------------------------------------------+
| Command          | Description                                                           |
+------------------+-----------------------------------------------------------------------+
| \d               | list tables                                                           |
| \d name          | describe table                                                        |
| \q               | quit datafusion-cli                                                   |
| \?               | help                                                                  |
| \format <format> | change output format [possible values: csv, tsv, table, json, ndjson] |
+------------------+-----------------------------------------------------------------------+
5 rows in set. Query took 0.005 seconds.
❯ \format json
❯ \?
[{"Command":"\\d","Description":"list tables"},{"Command":"\\d name","Description":"describe table"},{"Command":"\\q","Description":"quit datafusion-cli"},{"Command":"\\?","Description":"help"},{"Command":"\\format <format>","Description":"change output format [possible values: csv, tsv, table, json, ndjson]"}]
5 rows in set. Query took 0.001 seconds.
❯ \format invalid_format
invalid_format is not a valid format type [possible values: csv, tsv, table, json, ndjson]

```

# Are there any user-facing changes?

### CODE REVIEW

**Comment by capkurmagati** [comment]
@Jimexist PTAL.

File: datafusion-cli/src/command.rs [hunk 1]
```diff
@@ -34,6 +34,7 @@ pub enum Command {
     Help,
     ListTables,
     DescribeTable(String),
+    ChangeFomart(String),
```
Comment (Jimexist) [inline datafusion-cli/src/command.rs:37]:
```suggestion
    ChangeFormat(String),
```

**Review by Jimexist — COMMENTED** [review]
thanks! can you take a look at psql and see if there's any current convention for this command's naming?

**Comment by capkurmagati** [comment]
@Jimexist I took a look at the psql's formatting command. It looks like this.

```
Formatting
  \a                     toggle between unaligned and aligned output mode
  \C [STRING]            set table title, or unset if none
  \f [STRING]            show or set field separator for unaligned query output
  \H                     toggle HTML output mode (currently on)
  \pset [NAME [VALUE]]   set table output option
                         (border|columns|csv_fieldsep|expanded|fieldsep|
```

```
postgres=# \pset format
Output format is aligned.
postgres=# select 1 as a, 2 as b;
 a | b
---+---
 1 | 2
(1 row)

postgres=# \pset format csv
Output format is csv.
postgres=# select 1 as a, 2 as b;
a,b
1,2
postgres=# \pset format html
Output format is html.
postgres=# select 1 as a, 2 as b;
<table border="1">
  <tr>
    <th align="center">a</th>
    <th align="center">b</th>
  </tr>
  <tr valign="top">
    <td align="right">1</td>
    <td align="right">2</td>
  </tr>
</table>
<p>(1 row)<br />
</p>
```

Current implementation doesn't support multiple arguments yet. 
https://github.com/apache/arrow-datafusion/blob/master/datafusion-cli/src/command.rs#L110

One option is to use `\format` for this pr and create another issue to make `\` command support multiple arguments.
But also given it's a breaking change to deprecate `\format` in favor of `\pset format` later, to implement it at once seems reasonable.
WDYT.

**Comment by Jimexist** [comment]
> Current implementation doesn't support multiple arguments yet.

actually that's something on my mind to add

File: datafusion-cli/src/command.rs [hunk 2]
```diff
@@ -117,6 +126,7 @@ impl FromStr for Command {
             ("d", None) => Self::ListTables,
             ("d", Some(name)) => Self::DescribeTable(name.into()),
             ("?", None) => Self::Help,
+            ("format", Some(format)) => Self::ChangeFormat(format.into()),
```
Comment (Jimexist) [inline datafusion-cli/src/command.rs:129]: can you also add the command to print current format if no arg is provided?

File: datafusion-cli/src/exec.rs [hunk 3]
```diff
@@ -91,6 +91,13 @@ pub async fn exec_from_repl(ctx: &mut Context, print_options: &PrintOptions) {
                 if let Ok(cmd) = &line[1..].parse::<Command>() {
                     match cmd {
                         Command::Quit => break,
+                        Command::ChangeFormat(format) => {
+                            if let Ok(format) = PrintFormat::from_str(format) {
+                                print_options.format = format;
+                            } else {
+                                eprintln!("{} is not a valid format type [possible values: csv, tsv, table, json, ndjson]", format)
```
Comment (Jimexist) [inline datafusion-cli/src/exec.rs:98]: see https://github.com/apache/arrow-datafusion/pull/1284/files#r752332069

**Comment by Jimexist** [comment]
thanks for working on this. I have [one advice](https://github.com/apache/arrow-datafusion/pull/1284/files#r752332069) but feel free to add later?

**Comment by Jimexist** [comment]
hi @capkurmagati can you rebase again?

**Comment by capkurmagati** [comment]
@Jimexist Sorry being unresponsive. Let me reflect your comment on this weekend.
I think it's better to implement the command like Postgres, namely `\pset format`. 
But I don't have a good idea to organize the subcommands. To be specific, should we treat the subcommand(`pset format`) as an enum of `Command` or keep `pset` as an enum of `Command` and implement the subcommand logic in `Command::exec`. Do you have any thought on this?

**Comment by Jimexist** [comment]
> @Jimexist Sorry being unresponsive. Let me reflect your comment on this weekend.
> I think it's better to implement the command like Postgres, namely `\pset format`.
> But I don't have a good idea to organize the subcommands. To be specific, should we treat the subcommand(`pset format`) as an enum of `Command` or keep `pset` as an enum of `Command` and implement the subcommand logic in `Command::exec`. Do you have any thought on this?

I'd view `pset` as a command and the types of properties that you can set as a separate enum

**Review by Jimexist — APPROVED** [review]

**Pull request merged**